src/hangar/dataset.py

```diff
@@ -366,20 +366,12 @@ def __init__(self, stagehashenv, default_schema_backend, *args, **kwargs):
 
     def __enter__(self):
         self._is_conman = True
-        self._hashTxn = self._TxnRegister.begin_writer_txn(self._hashenv)
-        self._dataTxn = self._TxnRegister.begin_writer_txn(self._dataenv)
-        self._stageHashTxn = self._TxnRegister.begin_writer_txn(self._stagehashenv)
-        for k in self._fs.keys():
-            self._fs[k].__enter__()
+        self._cm_setup()
```

Comment:
@rlizzo I have a gut feeling that you are not going to like this. But I am not really a big fan of calling `__enter__` & `__exit__` from outside :(